src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/InstalledPackageReferenceIdComparer.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.CommandLine.XPlat
+{
+    class InstalledPackageReferenceIdComparer : IEqualityComparer<InstalledPackageReference>
```

Comment:
The comparer only compares packages names, and is used to keep unique entries for a dictionary.
If I were to override `Equals` for `InstalledPackageReference` and just use the name to compare, that would be a wrong behavior since 2 packages can have the same name, but different versions for example, so they are not really equal.